### ISSUE TITLE
TestBasic2B cannot find the region for the key

### ISSUE THREAD

**Issue by RinChanNOWWW** [talent-plan/tinykv#281]
```
=== RUN   TestBasic2B
2021/08/22 15:51:00.763986 node.go:193: [info] start raft store node, storeID: 1
2021/08/22 15:51:00.764297 peer.go:41: [info] region id:1 region_epoch:<conf_ver:1 version:1 > peers:<id:3 store_id:3 > peers:<id:4 store_id:4 > peers:<id:5 store_id:5 > peers:<id:1 store_id:1 > peers:<id:2 store_id:2 >  create peer with ID 1
2021/08/22 15:51:00.764408 raftstore.go:174: [info] start store 1, region_count 1, tombstone_count 0, takes 216µs
2021/08/22 15:51:00.766031 node.go:193: [info] start raft store node, storeID: 2
2021/08/22 15:51:00.766213 peer.go:41: [info] region id:1 region_epoch:<conf_ver:1 version:1 > peers:<id:3 store_id:3 > peers:<id:4 store_id:4 > peers:<id:5 store_id:5 > peers:<id:1 store_id:1 > peers:<id:2 store_id:2 >  create peer with ID 2
2021/08/22 15:51:00.766285 raftstore.go:174: [info] start store 2, region_count 1, tombstone_count 0, takes 121.7µs
2021/08/22 15:51:00.766464 node.go:193: [info] start raft store node, storeID: 3
2021/08/22 15:51:00.766529 peer.go:41: [info] region id:1 region_epoch:<conf_ver:1 version:1 > peers:<id:3 store_id:3 > peers:<id:4 store_id:4 > peers:<id:5 store_id:5 > peers:<id:1 store_id:1 > peers:<id:2 store_id:2 >  create peer with ID 3
2021/08/22 15:51:00.766558 raftstore.go:174: [info] start store 3, region_count 1, tombstone_count 0, takes 49.3µs
2021/08/22 15:51:00.767489 node.go:193: [info] start raft store node, storeID: 4
2021/08/22 15:51:00.767612 peer.go:41: [info] region id:1 region_epoch:<conf_ver:1 version:1 > peers:<id:3 store_id:3 > peers:<id:4 store_id:4 > peers:<id:5 store_id:5 > peers:<id:1 store_id:1 > peers:<id:2 store_id:2 >  create peer with ID 4
2021/08/22 15:51:00.767663 raftstore.go:174: [info] start store 4, region_count 1, tombstone_count 0, takes 93µs
2021/08/22 15:51:00.768773 node.go:193: [info] start raft store node, storeID: 5
2021/08/22 15:51:00.768890 peer.go:41: [info] region id:1 region_epoch:<conf_ver:1 version:1 > peers:<id:3 store_id:3 > peers:<id:4 store_id:4 > peers:<id:5 store_id:5 > peers:<id:1 store_id:1 > peers:<id:2 store_id:2 >  create peer with ID 5
2021/08/22 15:51:00.768940 raftstore.go:174: [info] start store 5, region_count 1, tombstone_count 0, takes 91.9µs
2021/08/22 15:51:01.769794 test_test.go:209: [info] Iteration 0

2021/08/22 15:51:01.769878 test_test.go:229: [info] 0: client new scan 0 00000000-0 00000000

2021/08/22 15:51:01.769895 test_test.go:222: [info] 0: client new put 0 00000000,x 0 0 y

    test_test.go:44: failure
panic: find no region for 30203030303030303030
```

Is it because the codes I wrote in 2A and 2B cause the problem? I think my codes are irrelevant to `Region`.

**Comment by xishuaidelin** [comment]
hello have you solve this problem？

**Comment by RinChanNOWWW** [comment]
> hello have you solve this problem？

Yes. I think it was because I didn't recover the config state from `Storage.InitialState()` in `newRaft` when `Config.peer` is `nil`. 

I fixed this by https://github.com/RinChanNOWWW/tinykv-impl/commit/1ea50f472ddcebed70b6eee4109598e9168edc4c#diff-b9adbc46e4a317ffbb3d11a66c38d6b9af41a09170d77d87efbd96d115da452fR175 .

**Comment by xishuaidelin** [comment]
> > hello have you solve this problem？
> 
> Yes. I think it was because I didn't recover the config state from `Storage.InitialState()` in `newRaft` when `Config.peer` is `nil`.
> 
> I fixed this by [RinChanNOWWW/tinykv-impl@1ea50f4#diff-b9adbc46e4a317ffbb3d11a66c38d6b9af41a09170d77d87efbd96d115da452fR175](https://github.com/RinChanNOWWW/tinykv-impl/commit/1ea50f472ddcebed70b6eee4109598e9168edc4c#diff-b9adbc46e4a317ffbb3d11a66c38d6b9af41a09170d77d87efbd96d115da452fR175) .

ohhhhh!
